src/components/TextInput/TextInput.tsx

```diff
@@ -247,7 +309,16 @@ export const TextInput = chakra(
           }
         : {
             "aria-required": isRequired,
-            autoComplete: hasAutocomplete ? type : null,
+            /** If the `autoComplete` prop is passed, that value will take
+             * precedence and will be used here. Otherwise, a value will be set
+             * based on the `type` prop. Lastly, if `autoComplete` is not passed
+             * and a default value is not set based on the `type` prop, then
+             * `autoComplete` will not be set for the input. */
+            autoComplete: hasAutocomplete
+              ? autoComplete
+                ? autoComplete
+                : type
+              : null,
```

Comment:
@EdwinGuzman Would this be better as `undefined`?

Reply:
I think either `null` or `undefined` work for attributes we don't want to render. I thought I read that react recommends `undefined` but I can't find any info on that. So this should be okay.

Reply:
Ok, I will leave it as is.